app/models/user.rb

```diff
@@ -19,4 +21,42 @@ class User < ApplicationRecord
   validates :email, presence: true, length: { maximum: 255 },
                     format: { with: VALID_EMAIL_REGEX },
                     uniqueness: { case_sensitive: false }
+
+  def friends
+    friends = friendships.includes(:friend).where(confirmed: true).references(:users)
+    friends.map(&:friend)
+  end
+
+  def pending_friends
+    requests = friendships.includes(:friend).where(confirmed: false).references(:users)
+    requests.map(&:friend)
+  end
+
+  # Users who have requested to be friends
+  def friend_requests
+    inverse_friendships.map { |friendship| friendship.user unless friendship.confirmed }.compact
+  end
+
+  def send_request(friend)
+    return if Friendship.exists?(user_id: id, friend_id: friend.id) || friend.id == id
+
+    Friendship.create(user_id: id, friend_id: friend.id)
+  end
+
+  def confirm_friend(user)
+    friendship = inverse_friendships.find { |friend| friend.user == user }
+    friendship.confirmed = true
+    friendship.save
+
+    friendships.create(friend_id: user.id, confirmed: true)
+  end
```

Comment:
Please, move it into `Friendship` model

Reply:
I tried to move the method to the friendship model and did some refactoring, but it didn't work and I thought since this method has to handle the current_user confirming a request I think it is also okay to leave it in the user model. Thanks for your support I really appreciate it.